amr-wind/turbulence/RANS/KOmegaSST.cpp

```diff
@@ -182,24 +183,55 @@ void KOmegaSST<Transport>::update_turbulent_viscosity(const FieldState fstate)
 
                     f1_arr(i, j, k) = tmp_f1;
 
-                    diss_arr(i, j, k) = -beta_star * rho_arr(i, j, k) *
-                                        tke_arr(i, j, k) * sdr_arr(i, j, k);
-                    tke_lhs_arr(i, j, k) = 0.5 * beta_star * rho_arr(i, j, k) *
-                                           sdr_arr(i, j, k) * deltaT;
-
+                    // For TKE equation:
                     shear_prod_arr(i, j, k) = amrex::min<amrex::Real>(
-                        mu_arr(i, j, k) * tmp4 * tmp4,
+                        amrex::max<amrex::Real>(
+                            mu_arr(i, j, k) * tmp4 * tmp4, 0.0),
                         10.0 * beta_star * rho_arr(i, j, k) * tke_arr(i, j, k) *
                             sdr_arr(i, j, k));
 
-                    sdr_lhs_arr(i, j, k) = 0.5 * rho_arr(i, j, k) * beta *
+                    diss_arr(i, j, k) = -beta_star * rho_arr(i, j, k) *
+                                        tke_arr(i, j, k) * sdr_arr(i, j, k);
+
+                    tke_lhs_arr(i, j, k) = 0.5 * beta_star * rho_arr(i, j, k) *
                                            sdr_arr(i, j, k) * deltaT;
-                    sdr_src_arr(i, j, k) =
-                        rho_arr(i, j, k) * alpha * tmp4 * tmp4 +
-                        (1.0 - tmp_f1) * 2.0 * rho_arr(i, j, k) * sigma_omega2 *
-                            gko / (sdr_arr(i, j, k) + 1e-15);
-                    sdr_diss_arr(i, j, k) = -rho_arr(i, j, k) * beta *
-                                            sdr_arr(i, j, k) * sdr_arr(i, j, k);
+
+                    // For SDR equation:
+                    if (diff_type == DiffusionType::Crank_Nicolson) {
```

Comment:
There's a bunch of duplicate code in these if conditions. Can we remove that duplication? I can see ternary expressions helping here.

Reply:
Issue has been fixed.